lib/parliament-flow.js

```diff
@@ -100,30 +107,55 @@ INSERT {
 }
 
 /**
- * @param {string} parliamentFlowUri
+ * @param {Partial<ParliamentFlow>[] | Partial<ParliamentFlow>} flows
  * @param {string} statusUri
  * @param {boolean} updateSudo
  */
-async function updateParliamentFlowStatus(parliamentFlowUri, statusUri, isSudo=false) {
+async function updateParliamentFlowStatus(flowOrFlows, statusUri, isSudo=false) {
+  let flows;
+  if (!Array.isArray(flowOrFlows)) {
+    flows = [flowOrFlows] 
+  } else {
+    flows = flowOrFlows
+  }
+  if (
+    flows.some(
+      (flow) =>
+        !flow.hasOwnProperty("uri") && !flow.hasOwnProperty("parliamentId")
+    )
+  ) {
+    throw new Error("Parliament flow needs parliamentId or uri property");
+  }
   const queryString = `
-PREFIX dossier: <https://data.vlaanderen.be/ns/dossier#>
-PREFIX parl: <http://mu.semte.ch/vocabularies/ext/parlement/>
-PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
-PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
-PREFIX adms: <http://www.w3.org/ns/adms#>
+${prefixHeaderLines.adms}
+${prefixHeaderLines.parl}
 
 DELETE {
   GRAPH ${sparqlEscapeUri(VP_GRAPH_URI)} {
-    ${sparqlEscapeUri(parliamentFlowUri)} adms:status ?oldStatus .
+    ?flowUri adms:status ?oldStatus .
   }
 }
 INSERT {
   GRAPH ${sparqlEscapeUri(VP_GRAPH_URI)} {
-    ${sparqlEscapeUri(parliamentFlowUri)} adms:status ${sparqlEscapeUri(statusUri)} .
+    ?flowUri adms:status ${sparqlEscapeUri(statusUri)} .
+  }
+} 
+WHERE {
+  VALUES (?flowUri ?parliamentId) {
+    ${flows
+      .map(
+        ({uri, parliamentId}) =>
+          `(${uri ? sparqlEscapeUri(uri) : "UNDEF"} ${
+            parliamentId ? sparqlEscapeString(parliamentId) : "UNDEF"
+          })`
+      )
+      .join("\n    ")}
   }
-} WHERE {
   GRAPH ${sparqlEscapeUri(VP_GRAPH_URI)} {
-    ${sparqlEscapeUri(parliamentFlowUri)} adms:status ?oldStatus .
+    ?flowUri parl:id ?parliamentId .
```

Comment:
Just for my own clarity, since this looks weird at first, with `?flowUri` or `?parliamentId` possibly being `UNDEF`.

The way this works is that if this is called from the "old" `syncFlowsByStatus` (still used by `/debug-resync-error-flows`), the `?parliamentId` will be `UNDEF`. If it's called from the new `syncSubmittedFlows`, the `?flowUri` will be `UNDEF`. 

`UNDEF` basically works as "any value, even undefined for now".
However, since we have this line `?flowUri parl:id ?parliamentId .`, it will make sure that both variables are always bound, causing the DELETE/INSERT to work.

Clever solution!